### ISSUE TITLE
flutter' app crash met with gomobile package .aar

### ISSUE THREAD

**Issue by hjlp20150803** [flutter/flutter#15925]
## Steps to Reproduce

flutter' app crash met with gomobile package .aar

I follow the guidence https://flutter.io/platform-channels/ ... It works; I can see the result in Simulator.

I write a simple code in Golang, Gomobile bind -target=android, get .aar file, put into libs, the app stopped when the call happened. The code is very simple:

package bond

import (

)

func FirstCall() int{
return 20180311
}

func SecondCall() string{
return "20180311"
}

func ThirdCall() {
}

type Counter struct {
Value int
}

func (c *Counter) Inc() { c.Value++ }

func NewCounter() *Counter { return &Counter{ 5 } }
3) I create app with Android Studio, put the .aar file in, and call, it gets the right return.

final String stext=Bond.secondCall();
4) So the problem is, there are 2 bridges here; each of them work, but put together, it crashes.

Anybody could give me a solution? Is the problem in Flutter or Gomobile?

## Logs

Run your application with `flutter run` and attach all the log output.
xxxxx$ flutter run
Using hardware rendering with device Android SDK built for x86. If you get graphics artifacts,consider enabling software rendering with "--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...                                       0.9s
Resolving dependencies...                                    2.9s
Running 'gradlew assembleDebug'...
Configuration 'compile' in project ':app' is deprecated. Use 'implementation' instead.
Built build/app/outputs/apk/debug/app-debug.apk (26.5MB).
Installing build/app/outputs/apk/app.apk...                  1.3s
I/FlutterActivityDelegate( 8995): onResume setting current activity to this
D/        ( 8995): HostConnection::get() New Host Connection established 0x95e73240, tid 9011
D/EGL_emulation( 8995): eglMakeCurrent: 0xa4786ba0: ver 3 0 (tinfo 0xa47f1ef0)
Syncing files to device Android SDK built for x86...
I/Choreographer( 8995): Skipped 62 frames!  The application may be doing too much work on its main thread.
D/EGL_emulation( 8995): eglMakeCurrent: 0xaaf057e0: ver 3 0 (tinfo 0x97b7b110)

🔥  To hot reload your app on the fly, press "r". To restart the app entirely, press "R".
An Observatory debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:8104/
For a more detailed help message, press "h". To quit, press "q".

Initializing hot reload...
Reloaded 468 of 484 libraries in 2,622ms.

Performing full restart...
Restarted app in 1,081ms.
D/channel ( 8995): getBatteryLevel
D/Thread  ( 8995): in thread,before call bond
F/libc    ( 8995): Fatal signal 6 (SIGABRT), code -6 in tid 9067 (Thread-4)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'google/sdk_google_phone_x86/generic_x86:7.1.1/NYC/4498512:userdebug/test-keys'
Revision: '0'
ABI: 'x86'
pid: 8995, tid: 9067, name: Thread-4  >>> com.example.blender <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
    eax 00000000  ebx 00002323  ecx 0000236b  edx 00000006
    esi 8c87f978  edi 8c87f920
    xcs 00000073  xds 0000007b  xes 0000007b  xfs 0000003b  xss 0000007b
    eip b44e9424  ebp 8c87f4e8  esp 8c87f48c  flags 00200296
backtrace:
    #00 pc ffffe424  [vdso:b44e9000] (__kernel_vsyscall+16)
    #01 pc 0007a03c  /system/lib/libc.so (tgkill+28)
    #02 pc 00075885  /system/lib/libc.so (pthread_kill+85)
    #03 pc 0002785a  /system/lib/libc.so (raise+42)
    #04 pc 0001ee36  /system/lib/libc.so (abort+86)
    #05 pc 000aff3a  /data/app/com.example.blender-2/lib/x86/libgojni.so
Lost connection to device.
xxxxx$
<img width="356" alt="stopped" src="https://user-images.githubusercontent.com/13619338/37876227-01060fda-307c-11e8-8964-a4515fda835d.png">

Run `flutter analyze` and attach any output of that command also.
xxxxx$ flutter analyze
Analyzing /Users/xxxxx/workbench/blender...
No issues found!
Ran in 6.9s
## Flutter Doctor

Paste the output of running `flutter doctor -v` here.
xxxxx$ flutter doctor -v
[✓] Flutter (Channel dev, v0.2.1, on Mac OS X 10.13.3 17D102, locale zh-Hans-CN)
    • Flutter version 0.2.1 at /Users/xxxxx/flutter
    • Framework revision 6f5bcb97a4 (2 weeks ago), 2018-03-08 15:01:37 -0800
    • Engine revision 24cf8ebd86
    • Dart version 2.0.0-dev.33.0.flutter-b5b8080b2c

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.2)
    • Android SDK at /Users/xxxxx/Library/Android/sdk
    • Android NDK at /Users/xxxxx/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.2
    • ANDROID_HOME = /Users/xxxxx/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2017.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 22.2.2
    • Dart plugin version 173.4548.30

[✓] VS Code (version 1.21.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.10.0

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

> For more information about diagnosing and reporting Flutter bugs, please see [https://flutter.io/bug-reports/](https://flutter.io/bug-reports/).

I tried lots of way to solve, it always shows "xx has stopped ",such call gomobile .arr as service,or in  new thread.
I am't not familiar with flutter or gomobile so deeply, I can't dig out the reason as you do,  I even can't tell whose responsibility is ,flutter or gomobile. I really hope flutter support golang as java,object C, or swift do ,or let golang as underneath language like c++ to develop package for dart. Such kind of work exceed of my skill, that maybe more easy for your flutter team  or golang team.
Since flutter is so good and let gomobile do same thing is not that smart thing, but let golang live with flutter maybe is a good choice.

**Comment by hjlp20150803** [comment]
The same thing could be done by react native with gomobile, why flutter can't ?
https://www.youtube.com/watch?v=OaSkOjew1D4&

**Comment by hjlp20150803** [comment]
I found the reason. First time I saw the '20180311' came out from screen,'20180311' is the response from gomobile package bond.aar. 
"gomobile bind -target=android",this is what I built the bond.aar, and my 'flutter doctor -v' was:
"[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)"
This is x86 type, so I change the gomobile to
"gomobile bind -target=android/x86"
and then put .aar into lib again, then it works.
I can't explain why they work independently, when the java program of android works and flutter project failed, but I solved the problem.
